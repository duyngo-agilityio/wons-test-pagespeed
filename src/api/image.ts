// Services
import { httpClient } from '@/services';

// utils
import { formatErrorMessage } from '@/utils';

export const uploadImage = async (
  image: FormData,
): Promise<string | { error: string }> => {
  try {
    const response = await httpClient.postRequest<
      FormData,
      { data: { url: string } }
    >({
      endpoint: `${process.env.NEXT_PUBLIC_UPLOAD_URL}?key=${process.env.NEXT_PUBLIC_UPLOAD_KEY}`,
      body: image,
      configOptions: {
        headers: {},
      },
    });

    return response.data.url;
  } catch (error) {
    const message = formatErrorMessage(error);
    return { error: message };
  }
};
