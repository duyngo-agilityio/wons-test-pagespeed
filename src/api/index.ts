export * from './invoice';
export * from './product';
export * from './statistic';
export * from './user';
export * from './customer';
export * from './task';
export * from './calendar';
export * from './image';
