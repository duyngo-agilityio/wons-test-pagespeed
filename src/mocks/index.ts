export * from './customer';
export * from './recentServices';
export * from './tabs';
