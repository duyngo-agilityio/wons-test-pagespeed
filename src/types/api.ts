export enum Method {
  Get = 'GET',
  Post = 'POST',
  Put = 'PUT',
  Patch = 'PATCH',
  Delete = 'DELETE',
}
