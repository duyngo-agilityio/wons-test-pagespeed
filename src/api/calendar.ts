// Constants
import { API_PATH } from '@/constants';

// Models
import { ICalendarTask, IEvent } from '@/models';

// Services
import { httpClient } from '@/services';

// Types
import { StrapiModel, StrapiResponse } from '@/types';

// Utils
import { formatErrorMessage } from '@/utils';

export const getCalendarEvents = async (): Promise<{
  error?: string;
  data?: StrapiModel<IEvent>[];
}> => {
  try {
    const productsResponse = await httpClient.getRequest<
      StrapiResponse<StrapiModel<IEvent>[]>
    >({
      endpoint: `${API_PATH.EVENTS}?populate=users_permissions_users`,
      configOptions: {
        next: {
          tags: [API_PATH.EVENTS],
        },
      },
    });

    return { data: productsResponse.data ?? [] };
  } catch (error) {
    const message = formatErrorMessage(error);
    return { error: message };
  }
};

export const getCalendarTasks = async (): Promise<{
  error?: string;
  data?: StrapiModel<ICalendarTask>[];
}> => {
  try {
    const productsResponse = await httpClient.getRequest<
      StrapiResponse<StrapiModel<ICalendarTask>[]>
    >({
      endpoint: `${API_PATH.CALENDAR_TASKS}`,
      configOptions: {
        next: {
          tags: [API_PATH.EVENTS],
        },
      },
    });

    if (!productsResponse?.data?.length) {
      return { data: [] };
    }

    return { data: productsResponse.data };
  } catch (error) {
    const message = formatErrorMessage(error);
    return { error: message };
  }
};
