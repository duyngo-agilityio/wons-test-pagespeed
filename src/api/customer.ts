// Constants
import { API_PATH } from '@/constants';
import { ICustomer } from '@/models';

// Services
import { httpClient } from '@/services';
import { StrapiModel, StrapiResponse } from '@/types';

// Utils
import { formatErrorMessage } from '@/utils';

export const getCustomers = async (): Promise<{
  error?: string;
  data?: StrapiModel<ICustomer>[];
}> => {
  try {
    const productsResponse = await httpClient.getRequest<
      StrapiResponse<StrapiModel<ICustomer>[]>
    >({
      endpoint: API_PATH.CUSTOMERS,
    });

    if (!productsResponse?.data?.length) {
      return { error: undefined, data: [] };
    }

    return { error: undefined, data: productsResponse.data };
  } catch (error) {
    const message = formatErrorMessage(error);
    return { error: message };
  }
};
