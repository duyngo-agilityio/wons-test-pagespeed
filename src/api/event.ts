// Constants
import { API_PATH } from '@/constants';

// Models
import { IEvent } from '@/models';

// Services
import { httpClient } from '@/services';

// Types
import { StrapiModel, StrapiResponse } from '@/types';

// Utils
import { formatErrorMessage } from '@/utils';

export const getEvents = async (): Promise<{
  error?: string;
  data?: StrapiModel<IEvent>[];
}> => {
  try {
    const productsResponse = await httpClient.getRequest<
      StrapiResponse<StrapiModel<IEvent>[]>
    >({
      endpoint: API_PATH.EVENTS,
      configOptions: {
        next: {
          tags: [API_PATH.EVENTS],
        },
      },
    });

    if (!productsResponse?.data?.length) {
      return { error: undefined, data: [] };
    }

    return { error: undefined, data: productsResponse.data };
  } catch (error) {
    const message = formatErrorMessage(error);
    return { error: message };
  }
};
