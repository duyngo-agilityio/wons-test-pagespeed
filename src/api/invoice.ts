// Constants
import { API_PATH, PAGE_SIZE } from '@/constants';

// Models
import { IProduct, TInvoiceProduct } from '@/models';

// Types
import { StrapiModel, StrapiResponse, TInvoiceListResponse } from '@/types';

// Utils
import { formatFilterIntervalDate } from '@/utils';

// Services
import { httpClient } from '@/services';

// Utils
import { formatErrorMessage } from '@/utils';

interface IParameters {
  cache?: RequestCache;
  nextOptions?: NextFetchRequestConfig;
  sort?: string;
  filters?: Record<string, string>;
}

export const getInvoiceProducts = async ({
  cache,
  nextOptions,
  sort,
  filters,
}: IParameters) => {
  const sortValue: string = sort ? `&sort=${sort}` : '';
  const filterQuery: string = formatFilterIntervalDate(
    filters as Record<string, string>,
  );

  const endpoint: string = `${API_PATH.INVOICE_PRODUCTS}?populate=product&
                            pagination[pageSize]=${PAGE_SIZE[4]}
                            ${sortValue}${filterQuery}`;
  try {
    const response = await httpClient.getRequest<
      StrapiResponse<StrapiModel<TInvoiceProduct<StrapiModel<IProduct>>>[]>
    >({
      endpoint: endpoint,
      configOptions: {
        cache: cache ?? 'force-cache',
        next: nextOptions,
      },
    });

    return response;
  } catch (error) {
    return error;
  }
};

export type InvoiceListConfigs = {
  query?: string;
  sortOrder?: string;
  sortBy?: string;
  cache?: RequestCache;
  nextOptions?: NextFetchRequestConfig;
};

export const getInvoices = async ({
  query,
  sortOrder,
  sortBy,
  cache,
  nextOptions,
}: InvoiceListConfigs) => {
  const sortValue = sortBy ? `&sort=${sortBy}:${sortOrder}` : '';
  const searchBy = query
    ? `&filters[$or][0][customer][fullName][$containsi]=${query}&filters[$or][1][customer][email][$containsi]=${query}`
    : '';
  const endpoint: string = `${API_PATH.INVOICES}?populate=customer${sortValue}${searchBy}`;

  try {
    const response = await httpClient.getRequest<TInvoiceListResponse>({
      endpoint,
      configOptions: {
        cache: cache ?? 'force-cache',
        next: nextOptions,
      },
    });

    return response;
  } catch (error) {
    const message = formatErrorMessage(error);

    throw new Error(message);
  }
};
