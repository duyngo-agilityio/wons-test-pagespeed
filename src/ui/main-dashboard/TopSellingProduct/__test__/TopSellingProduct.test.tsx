import { FunctionComponent } from 'react';
import { render, waitFor } from '@testing-library/react';

// components
import TopSellingProducts from '../index';

// apis
import { getAllProducts } from '@/api';

// mocks
import { mockProducts } from '@/mocks';

jest.mock('@/api', () => ({
  ...jest.requireActual('@/api'),
  getAllProducts: jest.fn(),
}));

async function resolvedComponent(
  Component: FunctionComponent<unknown>,
  props: unknown,
) {
  const ComponentResolved = await Component(props);
  return () => ComponentResolved;
}

describe('TopSellingProducts component', () => {
  it('renders correctly and matches snapshot', async () => {
    (getAllProducts as jest.Mock).mockResolvedValue({
      data: mockProducts,
    });

    const TopSellingProductsResolved = await resolvedComponent(
      TopSellingProducts,
      {},
    );
    const { container } = render(<TopSellingProductsResolved />);

    await waitFor(() => {
      expect(container).toMatchSnapshot();
    });
  });
});
