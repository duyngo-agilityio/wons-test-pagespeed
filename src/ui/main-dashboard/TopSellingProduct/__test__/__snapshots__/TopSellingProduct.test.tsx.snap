// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopSellingProducts component renders correctly and matches snapshot 1`] = `
<div>
  <div
    data-overlay-container="true"
  >
    <script>
      !function(){try{var d=document.documentElement,c=d.classList;c.remove('light','dark');var e=localStorage.getItem('theme');if('system'===e||(!e&&false)){var t='(prefers-color-scheme: dark)',m=window.matchMedia(t);if(m.media!==t||m.matches){d.style.colorScheme = 'dark';c.add('dark')}else{d.style.colorScheme = 'light';c.add('light')}}else if(e){c.add(e|| '')}else{c.add('light')}if(e==='light'||e==='dark'||!e)d.style.colorScheme=e||'light'}catch(e){}}()
    </script>
     
    <section
      class="bg-white dark:bg-gray-400 base:p-6 md:p-7 rounded-10 h-full md:!p-5"
    >
      <div
        class="flex justify-between mb-2"
      >
        <h2
          aria-level="2"
          class="font-bold font-dm-sans text-blue-800 dark:text-white text-2xl text-blue-800/70 dark:text-white/70"
        >
          Top Selling Products
        </h2>
      </div>
      <div
        class="mb-6 w-full"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 transition-transform-background motion-reduce:transition-none bg-white dark:bg-gray-400 rounded-lg shadow-none py-4"
          style="border-radius: undefined;"
          tabindex="-1"
        >
          <div
            class="flex flex-col sm:flex-row items-center"
          >
            <div
              class="w-24 h-24 sm:w-20 sm:h-20 mb-4 sm:mb-0"
            >
              <img
                alt="image alt"
                class="rounded-lg object-cover"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="http://localhost/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=256&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=256&q=75 2x"
                style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                width="100"
              />
            </div>
            <div
              class="sm:ml-4 flex-1 text-center sm:text-left"
            >
              <h3
                class="text-black dark:text-white font-semibold text-base sm:text-lg"
              >
                Product 1
              </h3>
              <div
                class="flex justify-center sm:justify-start py-2"
              >
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
              </div>
              <p
                class="text-black dark:text-white text-base sm:text-lg font-semibold"
              >
                $
                100
              </p>
            </div>
          </div>
        </div>
        <div
          class="border-t border-gray-300 dark:border-gray-600 my-8"
        />
      </div>
      <div
        class="mb-6 w-full"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 transition-transform-background motion-reduce:transition-none bg-white dark:bg-gray-400 rounded-lg shadow-none py-4"
          style="border-radius: undefined;"
          tabindex="-1"
        >
          <div
            class="flex flex-col sm:flex-row items-center"
          >
            <div
              class="w-24 h-24 sm:w-20 sm:h-20 mb-4 sm:mb-0"
            >
              <img
                alt="image alt"
                class="rounded-lg object-cover"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="http://localhost/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=256&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=256&q=75 2x"
                style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                width="100"
              />
            </div>
            <div
              class="sm:ml-4 flex-1 text-center sm:text-left"
            >
              <h3
                class="text-black dark:text-white font-semibold text-base sm:text-lg"
              >
                Product 2
              </h3>
              <div
                class="flex justify-center sm:justify-start py-2"
              >
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
              </div>
              <p
                class="text-black dark:text-white text-base sm:text-lg font-semibold"
              >
                $
                200
              </p>
            </div>
          </div>
        </div>
        <div
          class="border-t border-gray-300 dark:border-gray-600 my-8"
        />
      </div>
      <div
        class="mb-6 w-full"
      >
        <div
          class="flex flex-col relative overflow-hidden h-auto text-foreground box-border outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 transition-transform-background motion-reduce:transition-none bg-white dark:bg-gray-400 rounded-lg shadow-none py-4"
          style="border-radius: undefined;"
          tabindex="-1"
        >
          <div
            class="flex flex-col sm:flex-row items-center"
          >
            <div
              class="w-24 h-24 sm:w-20 sm:h-20 mb-4 sm:mb-0"
            >
              <img
                alt="image alt"
                class="rounded-lg object-cover"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="http://localhost/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=256&q=75"
                srcset="/_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=128&q=75 1x, /_next/image?url=https%3A%2F%2Fimg-cdn.pixlr.com%2Fimage-generator%2Fhistory%2F65bb506dcb310754719cf81f%2Fede935de-1138-4f66-8ed7-44bd16efc709%2Fmedium.webp&w=256&q=75 2x"
                style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
                width="100"
              />
            </div>
            <div
              class="sm:ml-4 flex-1 text-center sm:text-left"
            >
              <h3
                class="text-black dark:text-white font-semibold text-base sm:text-lg"
              >
                Product 3
              </h3>
              <div
                class="flex justify-center sm:justify-start py-2"
              >
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
                <svg
                  class="text-pink-500"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 576 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M259.3 17.8L194 150.2 47.9 171.5c-26.2 3.8-36.7 36.1-17.7 54.6l105.7 103-25 145.5c-4.5 26.3 23.2 46 46.4 33.7L288 439.6l130.7 68.7c23.2 12.2 50.9-7.4 46.4-33.7l-25-145.5 105.7-103c19-18.5 8.5-50.8-17.7-54.6L382 150.2 316.7 17.8c-11.7-23.6-45.6-23.9-57.4 0z"
                  />
                </svg>
              </div>
              <p
                class="text-black dark:text-white text-base sm:text-lg font-semibold"
              >
                $
                200
              </p>
            </div>
          </div>
        </div>
      </div>
    </section>
    <div
      class="fixed top-4 right-4"
    />
  </div>
</div>
`;
